CRM/MembershipExtras/Form/PaymentPlanSettings.php

```diff
@@ -69,6 +75,37 @@ private function addDefaultProcessorField($defaultProcessorField) {
     );
   }
 
+  private function addCustomGroupsToExcludeField($field) {
+    $customGroups = ['' => ts('- select -')] + $this->getCustomGroupsToExcludeFieldOptions();
+
+    $this->add(
+      $field['html_type'],
+      $field['name'],
+      ts($field['title']),
+      $customGroups,
+      $field['is_required'],
+      $field['extra_attributes']
+    );
+  }
+
+  private function getCustomGroupsToExcludeFieldOptions() {
+    $customGroups = civicrm_api3('CustomGroup', 'get', [
+      'sequential' => 1,
+      'return' => ["id", "title"],
+      'extends' => ['IN' => ['Contribution', 'ContributionRecur']],
+      'options' => ['limit' => 0],
+    ]);
+
+    $customGroupsOptions = [];
+    if (!empty($customGroups['values'])) {
+      foreach ($customGroups['values'] as $customGroup) {
+        $customGroupsOptions[$customGroup['id']] = $customGroup['title'];
```

Comment:
In the far off chance someone adds  a custom group with the same title to both Contributions and Recurring Contributions,  it might make sense to include the extended entity's name here, like `$customGroupsOptions[$customGroup['id']] = $customGroup['extends'] . ': ' . $customGroup['title']`.

Reply:
good point, thanks man!